File: Programmers/Python/Code/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    basket = []
    for i in moves:
        for j in range(len(board)):
            if board[j][i-1] != 0:
                if basket != [] and board[j][i-1] == basket[-1]:
                    answer += 2
                    basket.pop(-1)
                else:
                    basket.append(board[j][i-1])
                board[j][i-1] = 0
                break
    return answer

File: Programmers/JavaScript/Code/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    var stack = [];

    for (var i = 0; i < moves.length; i++) {
        var x = moves[i] - 1;
        for (var j = 0; j < board.length; j++) {
            if (board[j][x] != 0) {
                if (stack[stack.length - 1] === board[j][x]) {
                    stack.pop();
                    answer += 2;
                } else {
                    stack.push(board[j][x]);
                }
                board[j][x] = 0;
                break;
            }
        }
    }
    return answer;
}

// test case

// board = [
//   [0, 0, 0, 0, 0],
//   [0, 0, 1, 0, 3],
//   [0, 2, 5, 0, 1],
//   [4, 2, 4, 4, 2],
//   [3, 5, 1, 3, 1],
// ];
// moves = [1, 5, 3, 5, 1, 2, 1, 4];

// solution(board, moves);
